### ISSUE TITLE
Jira Cloud integration

### ISSUE THREAD

**Issue by JKnino** [go-gitea/gitea#25852]
### Feature Description

Hi there,

as many others so our company would like to migrate to gitea, unf none of available git plugins for jira does not work properly...creating branch, PR ..is there any plan to make this happen, or is there any other way to get it working and fully integrated with Jira?

cheers
Jas

**Comment by lunny** [comment]
I think there are already some plugins in the marketplace? Have you tried them?

**Comment by JKnino** [comment]
yes, there are actually just two plugins[ smart plugin](https://marketplace.atlassian.com/apps/1220080/smart-git-integration-for-jira?hosting=cloud&tab=overview) and [Git integration for Jira](https://marketplace.atlassian.com/apps/4984/git-integration-for-jira?hosting=cloud&tab=overview)

but none of them provide fully integration as service but simple repo..and that limits it to readonly

![image-20230712-112459](https://github.com/go-gitea/gitea/assets/99396486/250fd65b-b918-4613-8950-04611ad864f2)
![image-20230712-115454](https://github.com/go-gitea/gitea/assets/99396486/c23d8fdc-ea5e-403b-ac03-e3161b66346d)

**Comment by mathieulepoutrebelgium** [comment]
Hi JKnino

PR are fully supported by Exalate, this is an integration solution. I have been using it for a while for this exact use-case and it works wonders.

**Comment by molequetu** [comment]
Hello all,

As we are lately starting using PR, we would like somehow to integrate Gitea to Jira and see the information of the PR in the JIRA issue itself. At this point we have some partial integration with git via fisheye, where we see a development panel with the commits and branches created, but no PR related information for this jira issue key. 

After searcing for a while, I see that there is not an addon provider oriented to Gitea but the only solutions are using the 2 plugins that connect git with JIRA as mentioned above, the Smart Git Integration and GIt for Jira by gitkraken

Unfortunately, because of the version we use of JIRA(7.4.3) server on-premise, the second one by gitkraken is not available, and the first one(Smart Git Integration) cannot get a trial license for some reason. But to be honestly I am not sure if this plugin tracks PR's too(not showing this in their plugin page but only branches, commits, merges) where fisheye gives us this information already.

Upgrading JIRA to cloud or never server version is not in company plans ATM, so I was wondering if there is any other way to integrate GITEA with JIRA so we can have an overview of the PR's opened and merged or closed to that JIRA. Furthermore, as a second enchament we would like to transition JIRA issue status regarding the state of the PR(e.g when PR opened update status to IN_REVIEW) in JIRA, when PR request changes update to IN_PROGRESS, when PR merged update to READY state etc)

Any thoughts / insights you can share ?

Thnx in advance

**Comment by appleboy** [comment]
@mathieulepoutrebelgium You mean the plugin: https://marketplace.atlassian.com/apps/1213645/exalate-jira-issue-sync-more ?

**Comment by appleboy** [comment]
I created https://github.com/appleboy/jira-action to handle automatic Jira Issue Status transitions. Everyone should give it a try. Currently, it supports the Jira Data Center Version.

**Comment by appleboy** [comment]
I wrote a blog post 『[Integrating Gitea with Jira Software Development Workflow][1]』

[1]: https://blog.wu-boy.com/2025/03/gitea-jira-integration-en/

**Comment by warasilapm** [comment]
> I wrote a blog post 『[Integrating Gitea with Jira Software Development Workflow](https://blog.wu-boy.com/2025/03/gitea-jira-integration-en/)』

Updated link. https://blog.wu-boy.com/2025/03/gitea-integrate-with-jira-issue-tracking-flow-en/

**Comment by Spenhouet** [comment]
There are two features we would be looking for in a Gitea-Jira Integration:
- In Jira issues the side menu with "branches" and "pull requests" should display branches and PRs linked to the respective Jira issue
- Every commit message is required to contain a valid Jira issue ID, if not, the Git server should reject the pushed commits.

The above mentioned [jira-action](https://github.com/appleboy/jira-action) does not seem to provide these features or specifically focus on this. It introduces some extra integration into our Jira workflow, which we do not want.

Above some Jira add-ons where mentioned. I checked the Atlassian marketplace:
- Smart Git: I could not find this add-on in the market place, links on Google are dead, might no longer exist.
- [Git Integration by GitKraken](https://marketplace.atlassian.com/apps/4984/git-integration-for-jira-azure-devops-github-gitlab?hosting=cloud&tab=overview): Does a lot of things we don't need and does not seem to do the simple things we would need.
- [Git Connector by Getint](https://marketplace.atlassian.com/apps/1236406/git-connector-for-jira-azure-devops-github-gitlab?hosting=cloud&tab=overview): Does seem to provide the first feature in my list.
- [Jigit](https://marketplace.atlassian.com/apps/1217129/jigit-github-gitlab-azure-devops-integration-for-jira?hosting=datacenter&tab=overview): Only available for Jira Server. We are on Jira Cloud.

It would be nice if Atlassian would provide support for a on-prem Gitea instance. Is there already a feature request on Atlassian's issue tracker?
